app/models/db/user.rb

```diff
@@ -53,4 +53,22 @@ def avatar_url(size: nil)
   def activity_status
     active? ? 'active' : 'inactive'
   end
+
+  # Prepends an user's array containing last viewed
+  # boards with the `id` of given `DB::Board` and
+  # saves the user record.
+  #
+  # @param board [DB::Board]
+  # return [Void]
+  def last_viewed_board=(board)
+    return unless board.id
+
+    board_id = board.id.to_s
+    return if recent_boards.first == board_id
+
+    recent_boards.delete(board_id)
+    recent_boards.prepend(board_id)
+    self.recent_boards = recent_boards.first(5)
+    save(validate: false)
```

Comment:
Nie podoba mi się zapisywanie w setterze. Na pierwszy rzut oka z zewnątrz nie domyśliłbym się, że to mi od razu zapisze w bazie i to bez walidacji.

```ruby
user.last_viewed_board = board
```

Lepiej moim zdaniem bardziej explicit kod pisać

```ruby
user.last_viewed_board = board
user.save(validate: false)
```

Albo jak ci nie wygodnie, że musisz pisać tego save zawsze, to napisz drugą metodę z zapisem, ale która z nazwy widać, że zapisuje

```ruby
def update_last_viewed_board(board)
  self.last_viewed_board = board
  save(validate: false)
end
```

Chyba najbardziej mi się podoba ta ostatnia opcja.